crates/bevy_ecs/src/system/function_system.rs

```diff
@@ -187,8 +193,8 @@ impl<Param: SystemParam> SystemState<Param> {
     /// by a [`Commands`](`super::Commands`) parameter to the given [`World`].
     /// This function should be called manually after the values returned by [`SystemState::get`] and [`SystemState::get_mut`]
     /// are finished being used.
-    pub fn apply(&mut self, world: &mut World) {
-        self.param_state.apply(world);
+    pub fn apply(&mut self, system_meta: &SystemMeta, world: &mut World) {
+        self.param_state.apply(system_meta, world);
```

Comment:
```suggestion
    pub fn apply(&mut self, world: &mut World) {
        self.param_state.apply(&self.meta, world);
```

Not sure about this since, the name field isn't ever populated for SystemState. But it definitely feels weird that you'd need to construct a SystemMeta to apply the buffers when using SystemState.